src/Poseidon/EntitiesList.hs

```diff
@@ -171,87 +129,48 @@ instance EntitySpec PoseidonEntity where
-        groupNamesStats     = nub [ grp | Group grp               <- map underlyingEntity entities]
-        simpleIndNamesStats = nub [ ind | Ind   (SimpleInd ind)   <- map underlyingEntity entities]
-        specificIndsStats   = nub [ ind | Ind   (SpecificInd ind) <- map underlyingEntity entities]
-        missingPacs         = map Pac                 $ requestedPacVers    \\ pacNameVer
-        missingGroups       = map Group               $ groupNamesStats     \\ groupNamesPac
-        missingSimpleInds   = map (Ind . SimpleInd)   $ simpleIndNamesStats \\ indNamesPac
-        missingSpecificInds = map (Ind . SpecificInd) $ specificIndsStats   \\ availableInds
-    in missingPacs ++ missingGroups ++ missingSimpleInds ++ missingSpecificInds
-
--- This function requires [IndividualInfo] to be from a clean package selection as prepared with filterToRelevantPackages!
--- | Result: fst is a list of unresolved duplicates, snd a simple list of integers for the simple single individuals
-resolveEntityIndices :: (EntitySpec a) => [a] -> [IndividualInfo] -> ([[(Int, IndividualInfo, [SelectionState])]], [Int])
-resolveEntityIndices entities xs =
-    let allFittingIndizes = conformingEntityIndices entities xs
-        groupsOfEqualNameIndividuals = resolveDuplicatesIfPossible $ groupByIndividualName allFittingIndizes
-        unresolvedDuplicates = filter (\x -> length x > 1) groupsOfEqualNameIndividuals
-        simpleSingles = sort $ map (\(i,_,_) -> i) $ concat $ filter (\x -> length x == 1) groupsOfEqualNameIndividuals
-    in (unresolvedDuplicates, simpleSingles)
-    where
-        conformingEntityIndices :: (EntitySpec a) => [a] -> [IndividualInfo] -> [(Int, IndividualInfo, [SelectionState])]
-        conformingEntityIndices ents inds =
-            filter (\(_,_,level) -> not $ null level) $ map (\(index, x) -> (index, x, indInfoConformsToEntitySpec ents x)) (zip [0..] inds)
-        groupByIndividualName :: [(Int, IndividualInfo, [SelectionState])] -> [[(Int, IndividualInfo, [SelectionState])]]
-        groupByIndividualName entityIndices =
-            entityIndices &
-                sortBy (\(_,IndividualInfo a _ _,_) (_,IndividualInfo b _ _,_) -> compare a b) &
-                groupBy (\(_,IndividualInfo a _ _,_) (_,IndividualInfo b _ _,_) -> a == b)
-        resolveDuplicatesIfPossible :: [[(Int, IndividualInfo, [SelectionState])]] -> [[(Int, IndividualInfo, [SelectionState])]]
-        resolveDuplicatesIfPossible = map onlyKeepSpecifics
-        onlyKeepSpecifics :: [(Int, IndividualInfo, [SelectionState])] -> [(Int, IndividualInfo, [SelectionState])]
-        onlyKeepSpecifics groupOfInds =
-            let highPrio = filter (\(_,_,level) -> any isSpecified level) groupOfInds
-            in if length xs > 1 && length highPrio == 1
-               then highPrio
-               else groupOfInds
+determineNonExistentEntities entities indInfos = do -- for loop over entities
```

Comment:
Gladly. The code is:

```Haskell
determineNonExistentEntities entities indInfos = do -- for loop over entities
    entity <- underlyingEntity <$> entities
    let indices = resolveEntityIndices [entity] indInfos
    True <- return $ null indices
    return entity
```

Two things happen:
1) We loop over all entities. This is done via `entity <- underlyingEntity <$> entities`. What appears on the right side of the arrow is a list. On the left side is a single item of the list.
2) We _filter_ our main loop for cases in which `null indices` is True. Such a filter is also a feature of list-comprehensions, btw. This works because the list-monad has `fail` implemented as a skip in the loop, so a Pattern-match fail in a list-monad simply skips the respective loop-iteration.

Reply:
Will do!

Reply:
Done.